deploy/common/k8sbased.sh

```diff
@@ -55,6 +55,12 @@ function hotload_binary {
   echo
 
   binary_path=$(realpath "$(git rev-parse --show-toplevel)/bin/linux_amd64/${local_name}")
+  # set custom source path, e.g. when StackRox source is mounted into kind.
+  if [[ -n "$ROX_LOCAL_SOURCE_PATH" ]]; then
+      echo "ROX_LOCAL_SOURCE_PATH is set to $ROX_LOCAL_SOURCE_PATH."
+      binary_path="$ROX_LOCAL_SOURCE_PATH/bin/linux_amd64/${local_name}"
```

Comment:
Hm, in l57 we already do a git invocation for computing `binary_path` and you are overwriting this in case `$ROX_LOCAL_SOURCE_PATH` is non-empty, shouldn't we then the git git-invocation only in case binary_path remains empty after this conditional, i.e. in the case `$ROX_LOCAL_SOURCE_PATH` is empty do skip unnecessary work?

Reply:
Hm, in this case I think the git invoaction doesn't make a difference. Would like to keep it as it is eventhough I agree with your feedback.
Testing it would require me  to retest the deployment path in both scenarios which would take considerable effort for minimal improvement.